allennlp/modules/token_vectorizers/embedding.py

```diff
@@ -1,19 +1,29 @@
 import gzip
 import logging
 
+from overrides import overrides
 import numpy
 import torch
 
-from allennlp.data.vocabulary import Vocabulary
+from allennlp.common import Params
 from allennlp.common.checks import ConfigurationError
+from allennlp.data import Vocabulary
+from allennlp.experiments import Registry
+from allennlp.modules import TokenVectorizer
+
 logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
-class Embedding(torch.nn.Module):
+@Registry.register_token_vectorizer("embedding")
+class Embedding(TokenVectorizer):
     """
-    A more featureful embedding module than the default in Pytorch.
-    Adds the ability to pre-specify the weight matrix, use a non-trainable embedding and
-    catches some ill-advised use cases, such as 1 dimensional embeddings.
+    A more featureful embedding module than the default in Pytorch.  Adds the ability to
+    pre-specify the weight matrix or use a non-trainable embedding.
+
+    Note that it is not recommended that you use this directly when building models.  Instead, you
```

Comment:
It's worse than that, though, because then you have to worry about the details of the text field output in your model code.  You really shouldn't be using the `Embedding` layer directly if you're using a `TextField`, but the default assumption for an NLP person using an NLP library would be to look for an `Embedding` layer.  I'm trying to head off that confusion.